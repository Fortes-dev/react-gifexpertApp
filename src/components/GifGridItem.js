import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {

    return (

        //En vez de class, se usa className en react
        <div className="card animate__animated animate__backInLeft">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
