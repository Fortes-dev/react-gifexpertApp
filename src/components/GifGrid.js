import React, { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category); 

    return (
        <> 
        <h3 className="animate__animated animate__pulse">{category}</h3>
        { loading && <p className="animate__animated animate__fadeOut">Loading...</p>}
        <div className="card-grid">
                {
                    data.map( img => 
                        (
                            <GifGridItem 
                                {...img}
                                key={img.id}/>
                        )
                    )
                }
           
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
