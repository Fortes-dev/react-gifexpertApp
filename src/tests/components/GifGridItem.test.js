import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"


describe('Test Suite GifGridItem', () => { 
    
    // let wrapper;

    // beforeEach( () => {
    //     wrapper = shallow(<GifGridItem />)
    // })

    const title = 'Un título';
    const url   = 'https://localhost/algo.jpg';

    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('Debe mostrar el componente correctamente', () => { 
        
        expect(wrapper).toMatchSnapshot();

     });

    test('Debe de tener un párrafo con el title', () => { 
         
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    })

    test('Debe de tener la imagen igual al url y alt de los props', () => { 
        
        const img = wrapper.find('img');
        const urlImg = img.prop('src');
        const altImg = img.prop('alt');

        expect(urlImg).toBe(url);
        expect(altImg).toBe(title);

    });

    test('Debe de tener animate_fadeIn', () => { 
        
        const div = wrapper.find('div');
        const animation = 'animate__backInLeft';

        expect(div.prop('className')).toContain(animation);

    })

 })