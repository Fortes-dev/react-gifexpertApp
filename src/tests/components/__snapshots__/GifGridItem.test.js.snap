// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
